Mustachio/Tokenizer.cs

```diff
@@ -51,117 +51,135 @@ private static CharacterLocation HumanizeCharacterLocation(string content, int c
             return retval;
         }
 
-        public static IEnumerable<TokenPair> Tokenize(string templateString)
+        public static TokenizeResult Tokenize(string templateString, ParsingOptions parsingOptions)
         {
             templateString = templateString ?? "";
             var matches = _tokenFinder.Matches(templateString);
-            var scopestack = new Stack<Tuple<string, int>>();
+            var scopeStack = new Stack<Tuple<string, int>>();
+            var sourceName = parsingOptions.SourceName;
 
             var idx = 0;
 
+            var tokens = new List<TokenTuple>();
             var parseErrors = new List<IndexedParseException>();
             int[] lines = null;
 
+            var lowPrecedenceExpanders = parsingOptions.TokenExpanders?.Where(e => e.Precedence == Precedence.Low).ToList() ?? new List<TokenExpander>();
+            var mediumPrecedenceExpanders = parsingOptions.TokenExpanders?.Where(e => e.Precedence == Precedence.Medium).ToList() ?? new List<TokenExpander>();
+            var highPrecedenceExpanders = parsingOptions.TokenExpanders?.Where(e => e.Precedence == Precedence.High).ToList() ?? new List<TokenExpander>();
+
             foreach (Match m in matches)
             {
                 //yield front content.
                 if (m.Index > idx)
                 {
-                    yield return new TokenPair(TokenType.Content, templateString.Substring(idx, m.Index - idx));
+                    tokens.Add(new TokenTuple(TokenType.Content, templateString.Substring(idx, m.Index - idx)));
+                }
+                if (highPrecedenceExpanders.Any(e => Regex.IsMatch(m.Value, e.RegEx)))
```

Comment:
In the `if` statement, call into the method below and return bool if it matched/was processed. This will avoid multiple iteration on the expanders. It can also be parameterized with the precedence level to reuse later.